api/app/Http/Controllers/Api/UploadController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Actions\Upload\UploadImageAction;
+use App\Actions\Upload\UploadImageRequest;
+use App\Http\Presenters\ProfileImagePresenter;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class UploadController extends ApiController
+{
+    private ProfileImagePresenter $profileImagePresenter;
+    private UploadImageAction $uploadImageAction;
+
+    public function __construct(
+        ProfileImagePresenter $profileImagePresenter,
+        UploadImageAction $uploadImageAction
+    ) {
+        $this->profileImagePresenter = $profileImagePresenter;
+        $this->uploadImageAction = $uploadImageAction;
+    }
+
+    public function store(Request $request): JsonResponse
```

Comment:
On frontend we have the next workflow: if user uploads new image we send request to file upload action and receive url as response which we apply to user object and then send it to the UpdateUserAction. Also Volodymyr asked me to send all user data every request and update them in DB despite changed they or not

Reply:
@aussie-m-mike Ok. Then if we are going to use this flow, we must not delete an old avatar right after file uploading, because a user can leave the page without saving.